db/tests/types/test_inference.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from sqlalchemy import Column, MetaData, Table
+from sqlalchemy import BOOLEAN, Numeric, NUMERIC, String, VARCHAR
+from sqlalchemy.schema import CreateSchema, DropSchema
+from db.engine import _add_custom_types_to_engine
+from db.types import base, inference, install
+
+TEST_SCHEMA = "test_schema"
+
+
+@pytest.fixture
+def engine_with_types(engine):
+    _add_custom_types_to_engine(engine)
+    return engine
+
+
+@pytest.fixture
+def temporary_testing_schema(engine_with_types):
```

Comment:
You're using this fixture in multiple files, I recommend putting it in `conftest.py` in the root of this folder so that you can reuse it.

Reply:
Same with other reused fixtures.

Reply:
My reasoning was that these fixtures use code from mathesar that is tested by test files with _no_ access to these fixtures.  (This makes sure we're not using fixtures in a way that invalidates the code they use.  Do you know of a way to separate those out (or make it difficult/obvious when you _shouldn't_ use a given fixture for testing?  In particular, I'd want to find some way to guarantee that these fixtures won't be used for testing the `db/types/install.py` file at some point in the future.

Reply:
You could organize the files differently so that any tests that can use these fixtures are in a single directory with the appropriate `conftest.py` and files that shouldn't use these fixtures are in a different directory.

Reply:
that seems like the obvious solution to me but I might be missing something.

Reply:
We'd either need to break the "test file directory structure mirroring the real file directory structure" convention, or reorganize the real files in a (to me) less logical way, then mirror it.  I'm okay with the former, I guess, but I tend towards letting code repeat in test files over obfuscating what the files are testing (though it's not too hard to figure out at this stage in the project).

Reply:
We could also put these fixtures in a non-default file (i.e., not a `conftest.py` file), then import it.  It's kind of non-standard, but I think it'd be less confusing than partially mirroring the directory structure of the tested files with the structure of the test files.

Reply:
Actually, I think that's probably my favorite solution.

Reply:
For that kind of reorganization, I think it'd merit its own PR to document what was done and why with the restructuring.  @kgodey, what do you think?

Reply:
@mathemancer I think importing a non-standard file makes sense, and doing it in a separate PR also makes sense. I was originally thinking you'd reorganize the real files as well, but I like your solution too.

Reply:
I'll double-check that the organization of the real files makes sense and adjust accordingly.